Aggregate finagle-postgres with finagle-postgres-shapeless

leonmaia opened the pull request:
Problem:
Because the projects were not aggregated we had to have separate release
processes and some logic to run both modules tests.

Solution:
Aggregate both modules.

Result:
Now we have a single process for releasing and testing both modules.

leonmaia commented:
@vkostyukov I noticed yesterday that we are missing finagle-postgres-shapeless release. We've been releasing it every time we release finagle-postgres even though there are no changes on it in the past 2 years. Which makes me think that we only do this to make its versions matching with finagle-postgres. What do you think?

cc/ @jeremyrsmith

vkostyukov commented:
@leonmaia My understanding was `sbt release` does runs on a root project, which aggregates both, finagle-postgres and finagle-postgres-shapeless. So it should release both at the same time.


Update: Oh I see finagle-postgres-shapeles isn't aggregated into root. Maybe we should aggregate it [here](https://github.com/finagle/finagle-postgres/blob/master/build.sbt#L78). This should also allow us to remove that `test in Test` statement at the at do the build.sbt file.

vkostyukov commented:
Hey @leonmaia! Do you want me to give my idea a try - try to aggregate finagle-shapeless under the root project?

leonmaia commented:
@vkostyukov I'll try to do this today, I'm having really busy weeks. I tried to fix when you commented but I had issues since shapeless depends on finagle-postgres, so I could not only aggregate them on the root project. But let me try this today and I'll comment here my findings/solution.

leonmaia commented:
@vkostyukov I'm with a lot of things going on and I had no time to look into this. I'll hopefully have some time next week. I'm working on sorting things out to give this project some love ❤️ .

leonmaia committed:
Aggregate main module with shapeless module

Problem:
Because the projects were not aggregated we had to have separate release
processes and some logic to run both modules tests.

Solution:
Aggregate both modules.

Result:
Now we have a single process for releasing and testing both modules.

leonmaia commented:
@vkostyukov I believe these are the changes that fit what we discussed. Would you mind having a look?

vkostyukov commented:
Yep, this looks great! @leonmaia
